src/tools/msvc.jam

```diff
@@ -1797,6 +1797,8 @@ local rule register-toolset-really ( )
     toolset.flags msvc.compile CFLAGS <warnings>on : /W3 ;
     toolset.flags msvc.compile CFLAGS <warnings>off : /W0 ;
     toolset.flags msvc.compile CFLAGS <warnings>all : /W4 ;
+    toolset.flags msvc.compile CFLAGS <warnings>extra : /W4 ;
+    toolset.flags msvc.compile CFLAGS <warnings>pedantic : /W4 ;
```

Comment:
This possibly could be [`/Wall`](https://docs.microsoft.com/en-us/cpp/preprocessor/compiler-warnings-that-are-off-by-default?view=vs-2017)